File: isSameTree.py
from typing import *

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        def same(a,b):
            if a==None and b==None:
                return True
            if a==None or b==None:
                return False
            if a.val!=b.val:
                return False
            return same(a.left,b.left) and same(a.right,b.right)
        return same(p,q)

File: java/isSameTree.java
import java.util.HashSet;
import java.util.Set;
import java.util.Arrays;
import java.util.Stack;
import java.util.Queue;
import java.util.LinkedList;
import java.util.List;
import java.util.ArrayList;
import java.util.Deque;

class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p==null) {
            return q==null;
        }
        if(q==null) {
            return false;
        }
        if(p.val!=q.val) {
            return false;
        }
        return isSameTree(p.left,q.left)&&isSameTree(p.right,q.right);
    }

    public static void main(String[] args) {
        Solution sl=new Solution();
        System.out.println();
    }
}

